README.md

```diff
@@ -32,3 +32,116 @@ zopen build
 Note that running `zopen build` against your development repo will NOT install the code into your
 environment.
 
+
+# Developer Notes
+
+## workflow
+
+Say you want to work on meta
+
+### update existing meta
+
+This will update the ~/zopen/usr/local/bin versions of the system binaries as part
+of the meta package.
+
+```
+$ zopen upgrade meta -y
+```
+
+### Create the work dev repos for meta and metaport
+
+This sets up the repos for work.  Note this workflow
+requires meta and metaport dirs to be named meta
+and metaport respectively.
+
+```
+$ cd ~/zopen/dev
+$ git clone git@github.com:ZOSOpenTools/meta.git
+$ git clone git@github.com:ZOSOpenTools/metaport.git
+```
+
+### Create a branch in the dev version of meta for work
+
+
+```
+$ cd ~/zopen/dev/meta
+$ git branch johndev
+$ git checkout johndev
+```
+
+### Create an environment script to setup for dev work
+
+
+
+```
+$ cd ~/zopen/dev
+$ cat << EOF > setenv.sh
+> export ZOPEN_META_DEV_ROOT=$HOME/zopen/dev/meta
+> export ZOPEN_META_BRANCH=johndev
+> EOF
+
+
+$ cat setenv.sh
+export ZOPEN_META_DEV_ROOT=/z/jd895801/zopen/dev/meta.official
+export ZOPEN_META_BRANCH=johndev
+$
+```
+
+Source the script to have environment variables set in current shell.
+
+```
+$ cd ~/zopen/dev
+$ . ./setenv.sh
+```
+
+QUESTION
+
+This would be if I wanted to work on zotsampleport?
+
+How does it map ZOT_SAMPLEPORT to zotsampleport as in git repo name?
```

Comment:
If you wanted to enhance/improve zotsampleport - likely because you made changes to zopen-build or because you want to test zopen-build better, then you would:
```
cd $HOME/zopen/dev
git clone git@github.com:ZOSOpenTools/zotsampleport.git
cd zotsampleport
$ git branch mymod
$ git checkout mymod

<make your changes>
git add .
git commit -m 'comment'
git push --set-upstream origin mymod
```

You now need to be able to test zotsampleport before you merge your branch:
```
cd $HOME/zopen/dev/metaport
export ZOPEN_ZOT_SAMPLEPORT_BRANCH=mymod
zopen build
```
This will now use your branch `mymod` that you pushed to the server when running `zopen_check_build` testcase in the `tests` directory, by using your branch `mymod` instead of `main` when cloning `zotsampleport`

Reply:
Since this is specific to enhancing the testing of zopen-build, it should perhaps go after the other more general description?

Reply:
Sounds good.  I'll revisit this patch tomorrow.  I'll use your suggestions.  Thanks for all the good info.  You make this fun.